device_backends/uio/src/UioAccess.cc

```diff
@@ -56,10 +56,11 @@ namespace ChimeraTK {
       throw ChimeraTK::logic_error("UIO: Read request exceeds device memory region");
     }
 
-    void* targetAddress = static_cast<uint8_t*>(_deviceUserBase) + address;
-
-    // Is inherently thread-safe
-    std::memcpy(data, targetAddress, sizeInBytes);
+    volatile int32_t* rptr = static_cast<volatile int32_t*>(_deviceUserBase) + address / sizeof(int32_t);
+    while(sizeInBytes >= sizeof(int32_t)) {
+      *data++ = *rptr++;
```

Comment:
I never know which operator has precedence. I guess this is ```*(data++) = *(rptr++)```?

Reply:
Please make it explicit.

Reply:
If you do that, it first increments, then assigns, skipping the first entry.

will do
```
*data = *rptr
data++; rptr++
```